plaidml2/edsl/edsl_test.py

```diff
@@ -435,6 +491,106 @@ def test_skip(self):
+      K = Tensor(LogicalShape(plaidml.DType.FLOAT32, [3, 3, 1]))
+      O = conv_1d(I,K)
+      program = Program('conv_1d', [O])
+      if USE_MLIR():
+        self.assertMultiLineEqual(  
+                str(program), '''#map0 = (d0, d1, d2, d3, d4) -> (d0, d1, d2)
+#map1 = (d0, d1, d2, d3, d4) -> (d0, d1 + d3, d4)
+#map2 = (d0, d1, d2, d3, d4) -> (d3, d4, d2)
+
+
+!fp32 = type tensor<!eltwise.fp32>
+module {
+  func @conv_1d(%arg0: tensor<3x3x1x!eltwise.fp32>, %arg1: tensor<1x224x3x!eltwise.fp32>) -> tensor<1x222x1x!eltwise.fp32> {
+    %cst = "eltwise.sconst"() {value = 0.000000e+00 : f64} : () -> !fp32
+    %0 = tile.cion add, mul, %cst, %arg1, %arg0 {sink = #map0, srcs = [#map1, #map2]} : !fp32, tensor<1x224x3x!eltwise.fp32>, tensor<3x3x1x!eltwise.fp32> -> tensor<1x222x1x!eltwise.fp32>
+    return %0 : tensor<1x222x1x!eltwise.fp32>
+  }
+}
+''')       
+      else:
+        self.assertMultiLineEqual(
+                str(program), '''function (
+  _X0[_X0_0, _X0_1, _X0_2],
+  _X1[_X1_0, _X1_1, _X1_2]
+) -> (
+  _X2
+) {
+  _X2[x0, x1, x4 : 1, 222, 1] = +(_X0[x0, x1 + x2, x3] * _X1[x2, x3, x4]);
+}
+''')
+
+    def test_conv_2d_dialated(self):
+      I = Tensor(LogicalShape(plaidml.DType.FLOAT32, [1, 224, 224, 1]))
+      K = Tensor(LogicalShape(plaidml.DType.FLOAT32, [3, 3, 1, 32]))
+      O = conv_2d_dialated(I,K)
+      program = Program('conv_2d_dialated', [O])
+      if USE_MLIR():
+        #print(str(program))
+        self.assertMultiLineEqual(  
+                str(program), '''#map0 = (d0, d1, d2, d3, d4, d5, d6) -> (d0, d1, d2, d3)
+#map1 = (d0, d1, d2, d3, d4, d5, d6) -> (d0, d1 + d4 * 2, d2 + d5 * 3, d6)
+#map2 = (d0, d1, d2, d3, d4, d5, d6) -> (d4, d5, d6, d3)
+
+
+!fp32 = type tensor<!eltwise.fp32>
+module {
+  func @conv_2d_dialated(%arg0: tensor<3x3x1x32x!eltwise.fp32>, %arg1: tensor<1x224x224x1x!eltwise.fp32>) -> tensor<1x220x218x32x!eltwise.fp32> {
+    %cst = "eltwise.sconst"() {value = 0.000000e+00 : f64} : () -> !fp32
+    %0 = tile.cion add, mul, %cst, %arg1, %arg0 {sink = #map0, srcs = [#map1, #map2]} : !fp32, tensor<1x224x224x1x!eltwise.fp32>, tensor<3x3x1x32x!eltwise.fp32> -> tensor<1x220x218x32x!eltwise.fp32>
+    return %0 : tensor<1x220x218x32x!eltwise.fp32>
+  }
+}
+''')       
+      else:
+        self.assertMultiLineEqual(
+                str(program), '''function (
+  _X0[_X0_0, _X0_1, _X0_2, _X0_3],
+  _X1[_X1_0, _X1_1, _X1_2, _X1_3]
+) -> (
+  _X2
+) {
+  _X2[x0, x1, x3, x6 : 1, 220, 218, 32] = +(_X0[x0, x1 + 2*x2, x3 + 3*x4, x5] * _X1[x2, x4, x5, x6]);
+}
+''')
+
+    def test_complex_conv_2d(self):
+      I = Tensor(LogicalShape(plaidml.DType.FLOAT32, [1, 1, 1, 1, 1]))
+      K = Tensor(LogicalShape(plaidml.DType.FLOAT32, [1, 1, 1, 1, 1]))
+      O = complex_conv_2d(I,K,1,2,1,2)
+      program = Program('complex_conv_2d', [O])
+      if USE_MLIR():
+        self.assertMultiLineEqual(  
+                str(program), '''#map0 = (d0, d1, d2, d3, d4, d5, d6, d7) -> (d0, d1, d2, d3, d4)
```

Comment:
Can we add a new line here but then call `trim` so that we aren't comparing against the extra newline?

Reply:
for all tests?

Reply:
fixed. Let me know if this is what you meant.

Reply:
Yes, but it'd be nice to use a helper function so we don't to put `lstrip` everywhere. Also, there's an extra newline that we don't need.